src/appleseed/foundation/math/ray.h

```diff
@@ -230,6 +235,19 @@ inline Ray<T, N>::Ray(const Ray<U, N>& rhs)
 }
 
 template <typename T, size_t N>
+inline typename Ray<T, N>::ValueType Ray<T, N>::get_length() const
+{
+    assert(is_finite());
+    return foundation::norm((m_tmax - m_tmin) * m_dir);
```

Comment:
Small optimization: take the norm of `m_dir` *then* multiply by `m_tmax - m_tmin`. This will avoid two multiplications.